src/common/containers/NewSet.tsx

```diff
@@ -0,0 +1,221 @@
+          display: 'flex',
+          justifyContent: 'space-between',
+          alignItems: 'flex-start'
+        }}>
+          <div style={{
+            margin: '30px 10px 50px 30px',
+            width: 300,
+            minWidth: 300
+          }}>
+            <Paper style={{padding: 20, marginBottom: 20}}>
+              <div style={{fontWeight: 100, fontSize: 28}}>Energy Curve</div>
+              <EnergyCurve cards={this.selectedCards} />
+            </Paper>
+
+            {this.renderSidebarControls()}
+          </div>
+
+          <div style={{marginTop: 10, width: '100%'}}>
+            <CardCollection
+              layout={layout}
+              cards={this.displayedCards}
+              selectedCardIds={selectedCardIds}
+              onSelection={this.handleSelectCards} />
+          </div>
+
+          <div style={{
+            margin: '30px 30px 50px 10px',
+            width: 300,
+            minWidth: 300
+          }}>
+            <Paper style={{padding: 20}}>
+              <ActiveDeck
+                isASet
+                id={setBeingEdited ? setBeingEdited.id : ''}
+                name={setBeingEdited ? setBeingEdited.name : ''}
+                cards={this.selectedCards}
+                loggedIn={!!user}
+                onRemoveCard={this.handleRemoveCard}
+                onSave={this.handleClickSaveSet} />
+            </Paper>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  private renderSidebarControls = (): JSX.Element => (
+    <Paper style={{padding: 20, marginBottom: 10}}>
+      <SearchControls onChange={this.set('searchText')} />
+
+      <LayoutControls
+        layout={this.state.layout}
+        onSetLayout={this.set('layout')} />
+
+      <SortControls
+        criteria={this.state.sortCriteria}
+        order={this.state.sortOrder}
+        onSetCriteria={this.set('sortCriteria')}
+        onSetOrder={this.set('sortOrder')} />
+
+      <FilterControls
+        onToggleFilter={this.toggleFilter}
+        onSetCostRange={this.set('costRange')} />
+    </Paper>
+  )
+
+  // this.set(key)(value) = this.setState({key: value})
+  private set = (key: keyof NewSetState, callback = noop) => (value: any) => {
+    this.setState({[key]: value} as any, callback);
+  }
+
+  private toggleFilter = (filter: 'robots' | 'events' | 'structures') => (_e: React.SyntheticEvent<any>, toggled: boolean) => {
+    this.setState((state) => ({
+      filters: Object.assign({}, state.filters, {[filter]: toggled})
+    }));
+  }
+
+  private handleSelectCards = (selectedCardIds: string[]) => {
+    this.setState({ selectedCardIds });
+  }
+
+  private handleRemoveCard = (id: string) => {
+    this.setState((state) => ({
+      selectedCardIds: state.selectedCardIds.filter((cardId) => cardId !== id)
+    }));
+  }
+
+  private handleClickSaveSet = (id: string | null, name: string, cardIds: string[]) => {
+    const { allCards, user, onSaveSet, history } = this.props;
+
+    if (!user) {
```

Comment:
Why do we let people even get to this function if they're not logged in? `loggedIn` is passed in to ActiveDeck after all

Reply:
Yeah, but we have to still typecheck that `user` isn't `null`. This is just for TypeScript's sake.